rust/src/action.rs

```diff
@@ -362,6 +362,14 @@ pub struct Format {
     options: Option<HashMap<String, String>>,
 }
 
+impl Format {
+    pub fn new(provider: String, options: Option<HashMap<String,String>>) -> Self {
```

Comment:
You can add [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html#doc-comments) later on the `new` method.